compiler/rustc_infer/src/infer/error_reporting/note_and_explain.rs

```diff
@@ -242,7 +242,9 @@ impl<T> Trait<T> for X {
                         }
                     }
                     (ty::Alias(ty::Projection | ty::Inherent, proj_ty), _)
-                        if !tcx.is_impl_trait_in_trait(proj_ty.def_id) =>
+                        if !tcx.is_impl_trait_in_trait(proj_ty.def_id)
+                            // proj ty is forbidden as the type of a const generic parameter, skip it
```

Comment:
This comment and the line below it have no obvious connection.

What does `ObligationCauseCode::WhereClause` have to do with "forbidden as the type of a const generic parameter"?